source/sparse_matrix_simd.h

```diff
@@ -608,10 +634,44 @@ namespace ryujin
       }
     }
 
-    for (std::size_t c = 0; c < n_indices; ++c)
-      for (unsigned int comp = 0; comp < n_components; ++comp)
-        exchange_buffer[n_components * c + comp] =
-            data[n_components * sparsity->indices_to_be_sent[c] + comp];
+    /*
+     * Copy all entries that we plan to send over to the exchange buffer.
+     * Here, we have to be careful with indices falling into the "locally
+     * internal" range that are stored in an array-of-struct-of-array type.
+     */
+
+    for (std::size_t c = 0; c < n_indices; ++c) {
+
+      const auto &[row, position_within_column] =
+          sparsity->entries_to_be_sent[c];
+
+      Assert(row < sparsity->n_locally_owned_dofs, dealii::ExcInternalError());
+
+      if (row < sparsity->n_internal_dofs) {
+        // go through vectorized part
+        const unsigned int simd_row = row / simd_length;
+        const unsigned int simd_offset = row % simd_length;
+        for (unsigned int d = 0; d < n_components; ++d)
+          exchange_buffer[n_components * c + d] =
+              data[(sparsity->row_starts[simd_row] +
+                    position_within_column * simd_length) *
+                       n_components +
+                   d * simd_length + simd_offset];
```

Comment:
This is necessary, so that I can "slice" correctly with `simd_offset` here (which is impossible to reconstruct from `sparsity_row_starts[simd_row] + position_within_column * simd_length` alone).

Reply:
Downside is of course that we now have to access `sparsity->row_starts[...]` for every entry we send... but we have to access the sparsity pattern already anyway for every `get_entry`/`get_tensor` access. So this shouldn't be noticable.